test/gen-testfiles

```diff
@@ -18,11 +23,17 @@ mockery -name Interface -print \
     |sed 's/type Interface/type Ringpop/g' \
     > "$mock_directory"/ringpop.go
 
-echo
-
 mockery -name NodeInterface -recursive -print \
     |sed 's/_m \*NodeInterface/_m \*SwimNode/g' \
     |sed 's/type NodeInterface/type SwimNode/g' \
     > "$mock_directory"/swim_node.go
 
 mockery -case=underscore -dir router -recursive -output "$mock_directory" -name ClientFactory
+
+# make a EventListenerMock for forward (can't use on from $mock_directory because of circular dependency)
+mockery -name EventListener -dir events -recursive -print \
+    |sed 's/package mocks/package forward/g' \
+    > "${0%/*}/../forward/mock_event_listener_test.go"
```

Comment:
It gets the path part of command line string, i.e. so that the resulting path will be relative to the script location and not the current working directory.

`$0` means the command line string; `${variable_name%/*}` is a [string manipulation](http://tldp.org/LDP/abs/html/string-manipulation.html) operation that strips the shortest match of `"/*"` from the back of `$VAR`.